### PR TITLE
[bugfix] Remove heartbeat when unregistering worker

### PR DESCRIPTION
**Pull request by fw42**
This fixes a bug that was introduced in https://github.com/resque/resque/commit/4bb44413a045fca78d25e22ad535b3f5ea97fab8 where old heartbeats would no longer get cleaned up when a worker is being shut down. By itself, this is not a big deal, but in combination with https://github.com/resque/resque/pull/1479, it can lead to a race condition where workers that re-use previous worker ids (because hostname+pid+queues happen to match those of a previous worker) get pruned.

Also move all the heartbeat methods to the DataStore class.

@dylanahsmith @Sirupsen

### CODE REVIEW

**Comment by Sirupsen** [comment]
LGTM

**Comment by dylanahsmith** [comment]
:+1:

File: lib/resque/data_store.rb [hunk 1]
```diff
@@ -84,6 +89,11 @@ def all_resque_keys
       end
     end
 
+    def server_time
+      time, _ = @redis.time
```
Comment (davegudge) [inline lib/resque/data_store.rb]: It looks like this change is generating the following deprecation warning:

> Passing 'time' command to redis as is; blind passthrough has been deprecated and will be removed in redis-namespace 2.0

Comment (davegudge) [inline lib/resque/data_store.rb]: I see the issue will be resolved by updating `redis-namespace`

https://github.com/resque/redis-namespace/pull/119

**Pull request merged**